File: LinkedList.java
/*
1.Creation - done
Insertion after node - list.insertAfterNode();
Insertion at the end - list.insertNode();
Insertion in the beginning - list.insertAtHead();
Deletion on last node - list.deleteLastNode();
Deletion of specific node - list.deleteNode();
Deletion of first node - list.deleteFirstNode();
Display list - list.display();
List Reverse - list.reverse();
Merge 2 linked list sorted
 */

import java.util.Scanner;

class Node
{
    int data;
    Node next;
    Node(int data)
    {
        this.data=data;
        this.next=null;
    }
}

public class LinkedList {
    static Scanner sc=new Scanner(System.in);
    static Node head=null;
    public static void main(String args[])
    {
        LinkedList list=new LinkedList();
        int ch=0;
        do
        {
            System.out.println("\n1.Insert at the beginning");
            System.out.println("2.Insert at the end");
            System.out.println("3.Insert after a specific node");
            System.out.println("4.Delete first node");
            System.out.println("5.Delete last node");
            System.out.println("6.Delete specific node");
            System.out.println("7.Reverse list");
            System.out.println("8.Display");
            System.out.println("9.EXIT");
            System.out.println("Enter your choice:");
            ch=sc.nextInt();
            switch(ch)
            {
                case 1:
                    list.insertAtHead();
                    list.display();
                    break;
                case 2:
                    list.insertNode();
                    list.display();
                    break;
                case 3:
                    list.insertAfterNode();
                    list.display();
                    break;
                case 4:
                    list.deleteFirstNode();
                    list.display();
                    break;
                case 5:
                    list.deleteLastNode();
                    list.display();
                    break;
                case 6:
                    list.deleteNode();
                    list.display();
                    break;
                case 7:
                    list.reverse();
                    list.display();
                    break;
                case 8:
                    list.display();
                    break;
                case 9:
                    break;

            }
        }while(ch!=9);

    }
    public void reverse()
    {
        System.out.println("Reverse");
        if(head==null)
        {
            System.out.println("List is Empty!");
            return;
        }
        Node temp=null,prev=null,curr=null;
        temp=head;
        prev=head;
        if(head.next.next!=null) {
            curr = head.next.next;
            temp=head.next;
        }
        else
        {
            temp=temp.next;
            temp.next=head;
            head.next=null;
            head=temp;
            return;
        }
        while(curr.next!=null)
        {

            temp.next=prev;

            prev=temp;
            temp=curr;
            curr=curr.next;

        }
        if(curr.next==null)
        {
            temp.next=prev;
            curr.next=temp;
            head.next=null;
            head=curr;
        }
    }


    public void deleteNode()
    {
        if(head==null)
        {
            System.out.println("List is Empty!");
            return;
        }
        System.out.println("Enter the node to delete");
        int i=sc.nextInt();
        Node temp=head;
        Node prev=null;
        while(temp.data!=i) {
            prev=temp;
            temp = temp.next;
        }
        if(prev==null)
        {
            head=temp.next;
            temp=null;
        }
        else
        {
            prev.next=temp.next;
            temp=null;
        }
    }
    public void deleteFirstNode()
    {

        if(head==null)
        {
            System.out.println("List is Empty!");
            return;
        }
        if(head.next==null)
            head=null;
        else
        {
            Node temp;
            temp=head;
            head=head.next;
            temp=null;
        }
    }
    public void deleteLastNode()
    {
        if(head==null)
        {
            System.out.println("List is Empty!");
            return;
        }
        Node temp=head,prev=null;
        while(temp.next!=null) {
            prev=temp;
            temp = temp.next;
        }
        if(prev==null)
            head=null;
        else
        {
            prev.next=null;
            temp=null;
        }
    }
    public void insertAtHead()
    {
        System.out.println("Enter data to insert at the beginning:");
        int i=sc.nextInt();
        if(head==null)
            head = new Node(i);
        else
        {
            Node n1=new Node(i);
            n1.next=head;
            head=n1;
        }

    }
    public void display()
    {

        if(head==null)
        {
            System.out.println("List is Empty!");
            return;
        }
        System.out.println("Display List:");
        Node temp;
        temp=head;
        do {
            System.out.print(temp.data+"-->");
            temp=temp.next;
        }while(temp!=null);
        System.out.println();
    }
    public void insertNode()
    {
        System.out.println("Enter the data to insert at the end:");
        int i=sc.nextInt();
        if(head==null)
            head = new Node(i);
        else
        {
            Node temp;
            temp=head;
            while(temp.next!=null)
                temp=temp.next;
            Node n1=new Node(i);
            temp.next=n1;
            n1.next=null;
        }
    }

    public void insertAfterNode()
    {
        System.out.println("Enter the node to insert data after it:");
        int i=sc.nextInt();
        System.out.println("Enter the data to insert:");
        int j=sc.nextInt();
        if(head==null)
        {
            System.out.println("List is Empty!");
            return;
        }
        Node temp;
        temp=head;
        while(temp.data!=i && temp.next!=null)
            temp=temp.next;
        if(temp.next==null && temp.data!=i)
        {
            System.out.println("Node not present!");
            return;
        }
        else
        {
            Node n1=new Node(j);
            n1.next=temp.next;
            temp.next=n1;
        }
    }
}

File: LinkedList.py
# -*- coding: utf-8 -*-
"""
Created on Sun Jul 26 18:52:32 2020

@author: SUPERNOVA2813

Creation
Display - llist.display() 
Insert node at the end - llist.insertAtEnd()
Insert node at the beginning - llist.insertAtBeginning()
Insert after node - llist.insertAfterNode()
Delete a node - llist.deleteNode()
Delete last node - llist.deleteLastNode()
Delete first node - llist.deleteFirstNode()
Reverse - llist.reverse()

"""


class Node:
    def __init__(self,data):
        self.data=data
        self.next=None
    
class LinkedList:
    def __init__(self):
        self.head=None
    
    def insertAtEnd(self):
        i=int(input("Enter the data to insert node at the end:"))
        if(self.head==None):
            self.head=Node(i)
            self.head.next=None
        else:
            n1=self.head
            while(n1.next!=None):
                n1=n1.next
            n2=Node(i)
            n1.next=n2
            n2.next=None
            
    def insertAtBeginning(self):
        i=int(input("Enter the data to insert node at the beginning:"))
        if(self.head==None):
            self.head=Node(i)
            self.head.next=None
        else:
            n1=Node(i)
            n1.next=self.head
            self.head=n1
            
    def insertAfterNode(self):
        if(self.head==None):
            print("List is empty!")
            return
        i=int(input("Enter the node after which to insert data:"))
                 
        n1=self.head
        while(n1.data!=i and n1.next!=None):
            n1=n1.next
        if(n1.data==i):
            j=int(input("Enter the data for the node:"))
            n2=Node(j)
            n2.next=n1.next
            n1.next=n2
        else:
            print("Data not found to insert node after.")
    
    
    def deleteLastNode(self):
        if(self.head==None):
            print("List is empty!")
            return
        n1=self.head;
        while(n1.next!=None): 
            prev=n1
            n1=n1.next
        prev.next=None
        del n1
        
            
    def deleteFirstNode(self):
        if(self.head==None):
            print("List is empty!")
            return
        n1=self.head
        self.head=self.head.next
        del n1
        
        
    def deleteNode(self):
        if(self.head==None):
            print("List is empty!")
            return
        i=int(input("Enter the node to delete:"))
        n1=self.head
        while(n1.data!=i and n1.next!=None):
            prev=n1
            n1=n1.next
        if(n1.data==i):
            prev.next=n1.next
            del n1
        else:
            print("Data not found to delete")
            return    
            
    def display(self):
        if(self.head==None):
            print("List is empty!")
            return
        else:
            print("\n");
            n1=self.head
            while(n1!=None):
                print(n1.data,end = "--> ")
                n1=n1.next  
                
    def reverse(self):
        if(self.head==None):
            print("List is empty!")
            return
        n1=self.head
        if(n1.next==None):
            return
        if(n1.next.next==None):
            n1=n1.next
            n1.next=self.head
            self.head.next=None
            self.head=n1
        else:
            curr=n1.next.next
            prev=n1
            n1=n1.next
            while(curr.next!=None):
                n1.next=prev
                prev=n1
                n1=curr
                curr=curr.next
            if(curr.next==None):
                n1.next=prev
                curr.next=n1
                self.head.next=None
                self.head=curr


llist=LinkedList()
ch=0
while(ch!=9): 
    print("\n")
    print("1.Insert at the beginning")     
    print("2.Insert at the end")  
    print("3.Insert after a specific node") 
    print("4.Delete first node") 
    print("5.Delete last node")
    print("6.Delete specific node")
    print("7.Reverse list")
    print("8.Display")
    print("9.EXIT")
    ch=int(input("Enter Your Choice:"))
    if ch==1:
        llist.insertAtBeginning()
        llist.display()
    elif ch==2:
        llist.insertAtEnd() 
        llist.display()
    elif ch==3:
        llist.insertAfterNode()
        llist.display()
    elif ch==4:
        llist.deleteFirstNode()
        llist.display()
    elif ch==5:
        llist.deleteLastNode()
        llist.display()
    elif ch==6:
        llist.deleteNode()
        llist.display()
    elif ch==7:
        llist.reverse()
        llist.display()
    elif ch==8:
        llist.display()
    elif ch==9:
        break;
    else:
        print("Unknown input. Kindly enter correct choice!")
        



 